File: pdf-rag/embeddings.py
"""Generate embeddings via LiteLLM → Ollama."""

from __future__ import annotations

import litellm

from config import EMBED_MODEL, OLLAMA_BASE_URL


def embed_texts(texts: list[str]) -> list[list[float]]:
    """
    Return one embedding vector per input text.

    Uses LiteLLM which routes to Ollama's /api/embeddings endpoint.
    Batches automatically if the provider supports it.
    """
    response = litellm.embedding(
        model=EMBED_MODEL,
        input=texts,
        api_base=OLLAMA_BASE_URL,
    )
    # response.data is a list of EmbeddingObject with .embedding
    return [item["embedding"] for item in response.data]


def embed_single(text: str) -> list[float]:
    return embed_texts([text])[0]

File: llm-agent-rag/packages/app/src/embeddings.ts
import { embedMany, embed } from "ai";
import logger from "./logger.js";
import { EMBED_MODEL } from "./config.js";
import { getEmbeddingModel } from "./providers.js";

export async function embedTexts(texts: string[]): Promise<number[][]> {
  logger.debug({ count: texts.length, model: EMBED_MODEL }, "embedding batch");
  const model = getEmbeddingModel();
  const { embeddings } = await embedMany({ model, values: texts });
  return embeddings;
}

export async function embedSingle(text: string): Promise<number[]> {
  const model = getEmbeddingModel();
  const { embedding } = await embed({ model, value: text });
  return embedding;
}
